File: app/src/main/java/org/sorz/lab/tinykeepass/autofill/EntrySelectFragment.java
package org.sorz.lab.tinykeepass.autofill;


import android.content.Context;
import android.os.Build;
import androidx.annotation.RequiresApi;
import android.view.View;

import org.sorz.lab.tinykeepass.BaseEntryFragment;
import org.sorz.lab.tinykeepass.R;

import de.slackspace.openkeepass.domain.Entry;


@RequiresApi(api = Build.VERSION_CODES.O)
public class EntrySelectFragment extends BaseEntryFragment {
    private EntrySelectActivity activity;

    public EntrySelectFragment() {
        // Required empty public constructor
    }

    public static EntrySelectFragment newInstance() {
        return new EntrySelectFragment();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        activity = (EntrySelectActivity) context;
    }

    @Override
    protected int getFragmentLayout() {
        return R.layout.fragment_entry_select;
    }

    @Override
    protected void onEntryClick(View view, Entry entry) {
        activity.onEntrySelected(entry);
    }

    @Override
    protected boolean onEntryLongClick(View view, Entry entry) {
        return false;
    }

}

File: app/src/main/java/org/sorz/lab/tinykeepass/autofill/EntrySelectFragment.kt
package org.sorz.lab.tinykeepass.autofill

import android.content.Context
import android.os.Build
import android.view.Menu
import android.view.MenuInflater
import android.view.View
import androidx.annotation.RequiresApi
import com.kunzisoft.keepass.database.element.Entry
import org.sorz.lab.tinykeepass.BaseEntryFragment
import org.sorz.lab.tinykeepass.R

@RequiresApi(api = Build.VERSION_CODES.O)
class EntrySelectFragment : BaseEntryFragment() {
    override val fragmentLayout = R.layout.fragment_entry_select

    override fun onEntryClick(view: View, entry: Entry) {
        (requireActivity() as EntrySelectActivity).onEntrySelected(entry)
    }

    override fun onEntryLongClick(view: View, entry: Entry): Boolean {
        return false
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)
        menu.findItem(R.id.action_search).expandActionView()
    }

    companion object {
        @JvmStatic
        fun newInstance(): EntrySelectFragment {
            return EntrySelectFragment()
        }
    }
}